panic: runtime error: index out of range [0] with length 0

So I am getting:
```
panic: runtime error: index out of range [0] with length 0

goroutine 8 [running]:
github.com/abhinavxd/youtube-live-chat-downloader/v2.ParseInitialData({_, _})
        /home/mte90/go/pkg/mod/github.com/abhinavxd/youtube-live-chat-downloader/v2@v2.0.1/yt_chat.go:259 +0x446
cli-stream-chat/pkg/provider.(*Youtube).Listen(0x8eb868, 0xc0000163e0, {0x7ffd0c49f6e2, 0x2b}, {0xc000070300, 0x3, 0x4}, 0xc000024180)
        /home/mte90/Desktop/kde/cli-stream-chat/pkg/provider/youtube.go:15 +0x66
created by main.runListeners
        /home/mte90/Desktop/kde/cli-stream-chat/chat.go:26 +0x21a
exit status 2
```

My guess is that the regex is not working and return null https://github.com/abhinavxd/youtube-live-chat-downloader/blob/main/yt_chat.go#L141

Ref: https://github.com/blacktrub/cli-stream-chat/issues/1

--- a/yt_chat.go
+++ b/yt_chat.go
@@ -5,6 +5,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"math/rand"
 	"net/http"
 	"regexp"
 	"strconv"
@@ -123,7 +124,7 @@ type InitialData struct {
 									SubMenuItems []SubMenuItems `json:"subMenuItems"`
 								}
 							}
-						}	
+						}
 					}
 				}
 			}
@@ -205,10 +206,10 @@ func fetchChatMessages(initialContinuationInfo string, ytCfg YtCfg) ([]ChatMessa
 				} else {
 					if run.Emoji.IsCustomEmoji {
 						numberOfThumbnails := len(run.Emoji.Image.Thumbnails)
-						// Youtube chat has custom emojis which 
+						// Youtube chat has custom emojis which
 						// are small PNG images and cannot be displayed as text.
 						//
-						// These custom emojis are available with their image url.						
+						// These custom emojis are available with their image url.
 						//
 						// Adding some whitespace around custom image URLs
 						// without the whitespace it would be difficult to parse these URLs
@@ -240,12 +241,38 @@ func fetchChatMessages(initialContinuationInfo string, ytCfg YtCfg) ([]ChatMessa
 }
 
 func ParseInitialData(videoUrl string) (string, YtCfg, error) {
-	resp, err := http.Get(videoUrl)
+	req, err := http.NewRequest("GET", videoUrl, nil)
+	if err != nil {
+		return "", YtCfg{}, err
+	}
+	rand.Seed(time.Now().UnixNano())
+	cookie := &http.Cookie{
+		Name:   "CONSENT",
+		Value:  fmt.Sprintf("YES+yt.432048971.it+FX+%d", 100+rand.Intn(999-100+1)),
+		MaxAge: 300,
+	}
+	req.AddCookie(cookie)
+
+	cookie2 := &http.Cookie{
+		Name:   "PREF",
+		Value:  "tz=Europe.Rome",
+		MaxAge: 300,
+	}
+	req.AddCookie(cookie2)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
 	if err != nil {
 		return "", YtCfg{}, err
 	}
 
 	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		fmt.Print(videoUrl +
+			"\nresp.StatusCode: " + strconv.Itoa(resp.StatusCode))
+		return "", YtCfg{}, err
+	}
 
 	intArr, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
@@ -256,7 +283,7 @@ func ParseInitialData(videoUrl string) (string, YtCfg, error) {
 
 	// TODO ::  work on regex
 	initialDataArr := regexSearch(INITIAL_DATA_REGEX, html)
-	initialData := strings.Trim(initialDataArr[0], "ytInitialData = ")
+	initialData := strings.Trim(string(initialDataArr[0]), "ytInitialData = ")
 	initialData = strings.Trim(initialData, ";</script")
 	ytCfg := regexSearch(YT_CFG_REGEX, html)[0]
 	ytCfg = strings.Trim(ytCfg, "ytcfg.set(")
@@ -269,6 +296,9 @@ func ParseInitialData(videoUrl string) (string, YtCfg, error) {
 	json.Unmarshal([]byte(initialData), &_initialData)
 
 	subMenuItems := _initialData.Contents.TwoColumnWatchNextResults.ConversationBar.LiveChatRenderer.Header.LiveChatHeaderRenderer.ViewSelector.SortFilterSubMenuRenderer.SubMenuItems
+	if len(subMenuItems) == 0 {
+		return "", YtCfg{}, fmt.Errorf("empty initial data, the stream might not be live")
+	}
 	initialContinuationInfo := subMenuItems[1].Continuation.ReloadContinuationData.Continuation
 	return initialContinuationInfo, _ytCfg, nil
 }

